great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -2358,19 +2358,34 @@ def expect_column_pair_values_to_be_in_set(
         if len(parsed_value_set) == 0:
             return False
 
-        conditions: List[BinaryExpression] = [
-            sa.and_(
+        try:
```

Comment:
Instead of a try except could we use a conditional around a hasattr check? (similar to the other files in this PR)